api/src/db/DynamoBusinessesDataClient.ts

```diff
@@ -0,0 +1,151 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import { ExecuteStatementCommand, QueryCommand, QueryCommandInput } from "@aws-sdk/client-dynamodb";
+import { DeleteCommand, DynamoDBDocumentClient, GetCommand, PutCommand } from "@aws-sdk/lib-dynamodb";
+import { unmarshall } from "@aws-sdk/util-dynamodb";
+import { migrateData } from "@db/config/dynamoDbConfig";
+import { BusinessesDataClient } from "@domain/types";
+import { Business } from "@shared/userData";
+
+export const DynamoBusinessesDataClient = (
+  db: DynamoDBDocumentClient,
+  tableName: string
+): BusinessesDataClient => {
+  const doMigration = async (data: any): Promise<Business> => {
+    const migratedData = migrateData(data);
+    await put(migratedData);
+    return migratedData;
+  };
+
+  const findByIndex = (
+    indexName: string,
+    keyConditionExpression: string,
+    expressionAttributeValues: { [key: string]: any }
+  ): Promise<Business | undefined> => {
+    const params: QueryCommandInput = {
+      TableName: tableName,
+      IndexName: indexName,
+      KeyConditionExpression: keyConditionExpression,
+      ExpressionAttributeValues: expressionAttributeValues,
+    };
+
+    return db
+      .send(new QueryCommand(params))
+      .then(async (result) => {
+        if (!result.Items || result.Items.length !== 1) {
+          return;
+        }
+        const item = unmarshall(result.Items[0]);
+        return await doMigration(item.data);
+      })
```

Comment:
[dust] I like to avoid throwing `await` within `.then()` just because I find it's less cognitive load for the reader. Not a super strong opinion though.

Reply:
Hey @alexander-littleton,

Thanks for the feedback! I initially used that approach for consistency with the existing code, but it’s a good call out. I've gone ahead and adjusted both the `DynamoUserDataClient` and `DynamoBusinessesDataClient` to follow the suggested approach.

Appreciate the review!